core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/branch/ValidationCleanupService.java

```diff
@@ -0,0 +1,75 @@
+package com.b2international.snowowl.core.branch;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.eclipse.core.runtime.jobs.ISchedulingRule;
+
+import com.b2international.snowowl.core.ServiceProvider;
+import com.b2international.snowowl.core.events.AsyncRequest;
+import com.b2international.snowowl.core.identity.User;
+import com.b2international.snowowl.core.jobs.JobRequests;
+import com.b2international.snowowl.core.jobs.SerializableSchedulingRule;
+import com.b2international.snowowl.core.validation.ValidationRequests;
+import com.google.common.base.Joiner;
+
+public class ValidationCleanupService {
+	
+	public static final SerializableSchedulingRule VALIDATION_ISSUE_CLEANUP_RULE = new SerializableSchedulingRule() {
+		
+		private static final long serialVersionUID = 1L;
+
+		@Override
+		public boolean isConflicting(ISchedulingRule rule) {
+			return rule == this;
+		}
+		
+		@Override
+		public boolean contains(ISchedulingRule rule) {
+			return rule == this;
+		}
+	};
+
+	public ValidationCleanupService() {
+	}
+	
+	public void removeStaleIssues(ServiceProvider context, String resourceURI) {
+		removeStaleIssues(context, List.of(resourceURI), Collections.emptyList());
+	}
+	
+	public void removeStaleIssues(ServiceProvider context, List<String> resourceURIs) {
+		removeStaleIssues(context, resourceURIs, Collections.emptyList());
+	}
+	
+	public void removeStaleIssues(ServiceProvider context, String resourceURI, List<String> previousAnalysisRunIds) {
+		removeStaleIssues(context, List.of(resourceURI), previousAnalysisRunIds);
+	}
+	
+	public void removeStaleIssues(ServiceProvider context, List<String> resourceURIs, List<String> previousAnalysisRunIds) {
+		AsyncRequest<Boolean> request;
+		
+		if (!previousAnalysisRunIds.isEmpty()) {
+			request = ValidationRequests.issues()
+					.prepareDelete()
+					.setResultIds(previousAnalysisRunIds)
+					.buildAsync();
+		} else {
+			request = ValidationRequests.issues()
+				.prepareDelete()
+				.setCodeSystemURIs(resourceURIs)
```

Comment:
What happens if the `resourceURIs` is null or empty?

Reply:
Nothing good, we shouldn't delete without resource uri filters.

Reply:
Yeah, but that's a flaw in the API. If the caller should only use one input parameter, then we should split this method into two separate implementations instead of merging them into one with if else blocks.